polygeist/lib/polygeist/Ops.cpp

```diff
@@ -2043,33 +2043,31 @@ class SelectI1Ext final : public OpRewritePattern<arith::SelectOp> {
 
   LogicalResult matchAndRewrite(arith::SelectOp op,
                                 PatternRewriter &rewriter) const override {
-    auto ty = op.getType().dyn_cast<IntegerType>();
-    if (!ty)
-      return failure();
-    if (ty.getWidth() == 1)
-      return failure();
-    IntegerAttr lhs, rhs;
-    Value lhs_v = nullptr, rhs_v = nullptr;
-    if (auto ext = op.getTrueValue().getDefiningOp<arith::ExtUIOp>()) {
-      lhs_v = ext.getIn();
-      if (lhs_v.getType().cast<IntegerType>().getWidth() != 1)
-        return failure();
-    } else if (matchPattern(op.getTrueValue(), m_Constant(&lhs))) {
-    } else
+    // Cannot extui i1 to i1, or i1 to f32
```

Comment:
Comment states that an i1 cannot be extended to an i1 ? Clarify?

Reply:
Suggest:
// Transformation applies only to select operation yielding a boolean value.

Reply:
This comment is copied from https://github.com/intel/llvm/blob/8ccbd90efa74b0b0a4dfe15024a1d64bc190de07/mlir/lib/Dialect/Arith/IR/ArithOps.cpp#L2110.

It is corresponding to the check `op.getType().isInteger(1)`, we cannot extui from a i1 to a i1, as they are the same width.